lib/Dialect/FIRRTL/LowerToRTL.cpp

```diff
@@ -133,6 +133,34 @@ static LogicalResult lower(firrtl::PadPrimOp op, ArrayRef<Value> operands,
   return success();
 }
 
+static LogicalResult lower(firrtl::CatPrimOp op, ArrayRef<Value> operands,
+                           ConversionPatternRewriter &rewriter) {
+  auto lhs = mapOperand(operands[0], op, rewriter);
+  auto rhs = mapOperand(operands[1], op, rewriter);
+  if (!lhs || !rhs)
+    return failure();
+
+  auto lhsWidth = lhs.getType().cast<IntegerType>().getWidth();
+  auto rhsWidth = rhs.getType().cast<IntegerType>().getWidth();
+
+  // TODO full argument type conversion
```

Comment:
What is the TODO for?

Reply:
TODO was about:
We have clear assertions about the `firrtl.cat` argument types.
![image](https://user-images.githubusercontent.com/511872/85329507-81d49a00-b487-11ea-846b-16f2001d6ffe.png)
Do we have assertions about `rtl.concat` argument types?

Reply:
Ah, no we don't.  Those should go in a verification hook on the firrtl.cat operation itself. You can add verification assertions like this by implementing the verifier hook on the CatPrimOp definition in tblgen.  E.g. see how `verifyConstantOp` does additional verification for the `firrtl.constant` node.

Reply:
fixed